### DISCUSSION TITLE
Select the first N rows

### DISCUSSION THREAD

**Discussion by cool-RR** [saulpw/visidata#1884] [Q&A] [answered]
Hey,

Today I loaded a sheet with 3 million rows. VisiData is too slow on such sheets, so I wanted to open only 100,000 rows. I didn't know how to do this, so I added an index column and did `z|` to select all the rows with `A < 100_000`. This didn't work well, as the selection was too slow, and when I pressed `"` it didn't capture all the rows. Any better way to accomlish this? 

Thanks,
Ram.

**Comment by saulpw** [comment, marked as answer]
`zr100000` to go to that row, then `zs` to select all rows above it.

Also you might want `random-rows` which will give you a proper random sample.

**Reply by cool-RR** [reply]
Perfect, thank you for both of these.

By the way, I tried asking ChatGPT 4 about this, and it made up a `g#` command which unfortunately didn't exist. But still it's cute that ChatGPT understand how VisiData commands are supposed to look. I hope that at some point we'll be able to train an LLM on all the VisiData code and documentation so it could answer these questions. I actually tried feeing ChatGPT the list of commands, but it could only eat a third of them before it reached the limit.

**Reply by geekscrapy** [reply]
Maybe the GPT was trying to decipher the mnemonics (#1872) as a new language and `g#` is what made sense to it :)

**Reply by anjakefala** [reply]
This is actually in the works, and live on discord: https://github.com/saulpw/ai. You can chat with it if you are on discord!

But it also still gives silly answers that demonstates some understanding, but are factually incorrect. =)
